### ISSUE TITLE
CI: Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L3/L4 Flow

### ISSUE THREAD

**Issue by gandro** [cilium/cilium#11705]
## CI failure

Hubble tests fail to initialize due to test app not becoming ready.

Stacktrace:
```
/home/jenkins/workspace/Cilium-PR-K8s-newest-kernel-4.9/k8s-1.18-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:425
test pods are not ready after timeout
Expected
    <*errors.errorString | 0xc00007c530>: {
        s: "timed out waiting for pods with filter -l zgroup=testapp to be ready: 4m0s timeout expired",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-K8s-newest-kernel-4.9/k8s-1.18-gopath/src/github.com/cilium/cilium/test/k8sT/hubble.go:145
```

Subsequent tests fail with `failed due to BeforeAll failure`:
- Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L3/L4 Flow
- Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L3/L4 Flow with hubble-relay
- Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L7 Flow
- Suite-k8s-1.18.K8sHubbleTest Hubble Observe Test L7 Flow with hubble-relay

https://jenkins.cilium.io/job/Cilium-PR-K8s-newest-kernel-4.9/402/
[67f9f89d_K8sHubbleTest_Hubble_Observe_Test_L3-L4_Flow.zip](https://github.com/cilium/cilium/files/4687872/67f9f89d_K8sHubbleTest_Hubble_Observe_Test_L3-L4_Flow.zip)

**Comment by gandro** [comment]
The pods are unavailable because cilium-cni cannot be loaded:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = [failed to set up sandbox container "abb98a4dbf322db8f9deedddd906f0a70517c9032f245d6c13eb10fe3eef1a73" network for pod "app3-76694d56c5-lqvg9": networkPlugin cni failed to set up pod "app3-76694d56c5-lqvg9_202005270740k8shubbletesthubbleobservetestl3l4flow" network: failed to find plugin "cilium-cni" in path [/opt/cni/bin], failed to clean up sandbox container "abb98a4dbf322db8f9deedddd906f0a70517c9032f245d6c13eb10fe3eef1a73" network for pod "app3-76694d56c5-lqvg9": networkPlugin cni failed to teardown pod "app3-76694d56c5-lqvg9_202005270740k8shubbletesthubbleobservetestl3l4flow" network: failed to find plugin "cilium-cni" in path [/opt/cni/bin]]
```

Looks like the same root cause as: https://github.com/cilium/cilium/issues/11497

**Comment by gandro** [comment]
Hopefully fixed in #11553
